app/models/goal.py

```diff
@@ -2,4 +2,29 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+
+    def to_dict(self):
+        return {
+            "id": self.goal_id,
+            "title": self.title
+        }
+    
+    def to_dict_with_task(self):
+        task_list = []
+        for task in self.tasks: 
+            task_list.append(task.to_dict_with_goal())
+        return {
+            "id": self.goal_id,
+            "title": self.title,
+            "tasks": task_list
+        }
```

Comment:
I'd recommend combining these two functions since they're returning almost the same thing. I think it's appropriate to simply have a conditional inside of `to_dict()` that determines if it should also return the list of tasks. This way, you don't need to manage calling two different functions in your routes.